### ISSUE TITLE
Add test helper for asserting component wasn't rendered

### ISSUE THREAD

**Issue by joelhawksley** [github/view_component#261]
While the Capybara matchers we have in this library are helpful, in practice there has been one test that is tricky to write: asserting that nothing was rendered for a component, often due to returning `false` for `render?`.

I'm thinking we need something like `refute_component_rendered`.

Right now, we have something like this:

```ruby
test "does not render unless the event is present" do
  render_inline(EventComponent.new(event: nil)

  assert page.text.empty?, "expected component to be empty"
end
```

But it's not foolproof. What if the component renders an icon, but not text? Ideally this assertion would ensure that _nothing_, not even a single DOM node, was rendered.

cc @kenyonj @nickh @BlakeWilliams

**Comment by BlakeWilliams** [comment]
Thanks for opening this!

I did some testing, and I *think* `assert_no_selector('body')` might actually work for this case since Capybara seems to render each component with a doctype, `html` tag, and `body` tag even if the component doesn't contain those tags. It will seemingly omit the `body` tag when nothing is rendered though.

It's not my favorite solution, but thought it was a discovery worth sharing.

**Comment by joelhawksley** [comment]
Closed by #263
